### ISSUE TITLE
react-native-camera iOS is not detecting any faces

### ISSUE THREAD

**Issue by St7-07** [react-native-community/react-native-camera#2349]
# Bug Report

**To Do First**

- [ YES ] Did you try latest release?
- [ YES ] Did you try master?
- [ YES ] Did you look for existing matching issues?

**Platforms**

**Versions**

- iOS: 12.3.1
- react-native-camera: ^2.11.0
- react-native: ^0.59.9
- react: ^16.8.6

**Description/Current Behaviour**

onFacesDetected is nos detecting any face, just return empty array. 
logs: Object {type: "face", faces: Array(0), target: 259}

Text Detector is working
onFaceDetectionError is never triggered 

**Expected Behaviour**

Detect faces when there is faces in the camera and return a filled array

**Steps to Reproduce**

Install react-native-camera via cocoa pods like 

  pod 'react-native-camera', path: '../node_modules/react-native-camera', subspecs: [
    'FaceDetectorMLKit', 'TextDetector'
  ]

**Additionals**

Code example:

      <RNCamera
        ref={(ref) => {
          this.camera = ref;
        }}
        captureAudio={false}
        style={styles.preview}
        type={cameraType} // already try with back and front, none of both works
        defaultVideoQuality={RNCamera.Constants.VideoQuality['720p']}
        permissionDialogTitle={i18n.t('KYCValidation_CameraAccessTitle')}
        permissionDialogMessage={i18n.t('KYCValidation_CameraAccessMessage')}
        onTextRecognized={({ textBlocks }) => console.log(textBlocks)}
        faceDetectionMode={RNCamera.Constants.FaceDetection.Mode.accurate} // fast doesn't work neither
        faceDetectionClassifications={RNCamera.Constants.FaceDetection.Classifications.none}
        onFacesDetected={faces => console.log(faces)}
        onFaceDetectionError={error => console.log('FDError', error)} // This is never triggered
      >

**Comment by eduardinni** [comment]
same problem with the same env

**Comment by demianp** [comment]
try this workaround:
onCameraReady={() => this.setState({canDetectFaces: true})}
onFacesDetected={this.state.canDetectFaces ? this.facesDetected : null}

**Comment by barnapcristian** [comment]
Hello GitHub, this is my first time answering ever here :)

Here's how I managed to make it work on iOS with React Native 0.60 with both cameras front and back. Although I'm still waiting on an official fix, maybe with my answer they can pin point the problem and fix it. So for now it's temporary fix.
The problem as far as I investigated relies in these 2 lines:

    CGRect croppedSize = AVMakeRectWithAspectRatioInsideRect(previewSize, boundingRect);
    CGImageRef croppedCGImage = CGImageCreateWithImageInRect(videoImage, croppedSize);

So the solution that I found is:

1. Open Xcode with your project
2. Find and open the linked React-Native-Camera folder
3. Find and open RNCameraUtils.m
![Screen Shot 2019-07-18 at 4 40 43 PM](https://user-images.githubusercontent.com/49002827/61466789-d075dc00-a97a-11e9-9e10-daaf0bf2ec56.png)

4. At the end of the file you'll find
    videoImage = [temporaryContext createCGImage:ciImage fromRect:boundingRect];
    CGRect croppedSize = AVMakeRectWithAspectRatioInsideRect(previewSize, boundingRect);
    CGImageRef croppedCGImage = CGImageCreateWithImageInRect(videoImage, croppedSize);
    UIImage *image = [[UIImage alloc] initWithCGImage: **croppedCGImage**];
    CGImageRelease(videoImage);
    CGImageRelease(croppedCGImage);
    return image;
5. Replace it with
    videoImage = [temporaryContext createCGImage:ciImage fromRect:boundingRect];
    CGRect croppedSize = AVMakeRectWithAspectRatioInsideRect(previewSize, boundingRect);
    CGImageRef croppedCGImage = CGImageCreateWithImageInRect(videoImage, croppedSize);
    UIImage *image = [[UIImage alloc] initWithCGImage:**videoImage**];
    CGImageRelease(videoImage);
    CGImageRelease(croppedCGImage);
    return image;
![Screen Shot 2019-07-18 at 4 42 37 PM](https://user-images.githubusercontent.com/49002827/61466971-2054a300-a97b-11e9-8fe7-02d22cc18aa9.png)

The problem seems to be the croppedCGImage. So if we change the **initWithCGImage:croppedCGImage**
                             with
**initWithCGImage:videoImage** 
                         it works!

**But keep in mind that it's a temporary fix and if you update RNC, this fix will disappear. So better to not update until RNC says it's fixed.** 

Have a great day!

**Comment by sibelius** [comment]
@barnapcristian  can you send your fix as a pull request?

**Comment by jordangrant** [comment]
@barnapcristian solution works (thanks so much) saved me time

**Comment by igor-bv** [comment]
Will someone provide a PR to fix this bug and merge into original repo?

**Comment by sibelius** [comment]
fixed here https://github.com/react-native-community/react-native-camera/pull/2611

**Comment by viimauer** [comment]
> fixed here #2611

Its working only in front camera for me

**Comment by sibelius** [comment]
@viimauer can you open a new issue for back camera?

can you dig in the code?
